evm_arithmetization/src/prover.rs

```diff
@@ -543,6 +544,15 @@ pub fn generate_all_data_segments<F: RichField>(
         };
     }
 
+    // We need at least two segments to prove a segment aggregation.
+    if all_seg_data.len() == 1 {
+        let dummy_seg = GenerationSegmentData {
+            registers_before: segment_data.registers_after,
+            ..segment_data
+        };
+        all_seg_data.push(dummy_seg);
+    }
```

Comment:
Sorry, I'm not sure I undeerstand the question. Do you mean why do we need `registers_before` to be equal to the previous segment's `registers_after`? Or why do we need `registers_after` to be equal to `registers_before` in the dummy segment?

Reply:
Why dummy segment has `registers_after = previous_registers_after`. But I guess this would go away with the new pruning right?

Reply:
That's because the dummy segment starts at a halting state already, so the registers don't change.
But it would indeed go away with the dummy segment at the start.